app/models/concerns/dirty_associations.rb

```diff
@@ -33,6 +33,7 @@ def dirty_has_many_associations(*args)
           #   end
           define_method "#{association_ids}=" do |collection|
             collection ||= [] # in API, #{association}_ids is converted to nil if user sent empty array
+            collection = [collection].flatten # in case we got single id, we ensure it's an array
```

Comment:
Using `Array(collection)` might be faster

Reply:
updated